### PR TITLE
Robustify CI

### PR DESCRIPTION
**Pull request by andrewpbrett**
#### What? Why?

Addresses @mkllnk's comment: https://github.com/openfoodfoundation/openfoodnetwork/pull/7168#pullrequestreview-617092086

#### What should we test?
Fast green build. I've verified that the number of tests run is unchanged based on these commits. 

#### Release notes

Further robustified the test suite

Changelog Category: Technical changes

### CODE REVIEW

File: lib/tasks/specs.rake [hunk 1]
```diff
@@ -1,5 +1,31 @@
 namespace :ofn do
   namespace :specs do
+    namespace :run do
+      def spec_folders
+        Pathname("spec/").children.select(&:directory?).map { |p|
+          p.split.last.to_s
+        }.reject { |p| %w(support factories javascripts performance).include?(p) }
```
Comment (mkllnk) [inline lib/tasks/specs.rake:7]:
```rb
["a", "b"] - ["b", "c"]
=> ["a"]
```

```suggestion
        } - %w(support factories javascripts performance)
```

Possibly better formatting.

**Review by mkllnk — APPROVED** [review]
Nice. I double-checked the numbers and this seems to catch one additional spec. Master ran 1550 specs and this PR ran 1551.

File: lib/tasks/specs.rake (hunk 1 above)
Comment (andrewpbrett) [reply lib/tasks/specs.rake]: Agreed, thanks!

**Review by mkllnk — APPROVED** [review]

File: lib/tasks/specs.rake [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+      def execute_rspec_for_pattern(pattern)
+        system "bundle exec rspec --profile --pattern \"#{pattern}\""
+      end
+
+      def execute_rspec_for_spec_folder(folder)
+        execute_rspec_for_pattern("spec/#{folder}/{,/*/**}/*_spec.rb")
+      end
+
+      def execute_rspec_for_spec_folders(folders)
+        folders = folders.join(",") if folders.is_a?(Array)
```
Comment (sauloperez) [inline lib/tasks/specs.rake:19]: I believe `folders = folders.join(",")` alone would have the same result.

**Review by sauloperez — APPROVED** [review]

File: lib/tasks/specs.rake (hunk 2 above)
Comment (sauloperez) [reply lib/tasks/specs.rake:19]: I leave this here because I think it's worth changing this :point_up: Then we can merge.

Comment (andrewpbrett) [reply lib/tasks/specs.rake:19]: I was guarding against the case where the argument was a string, like "foo,bar,baz", but given that this is such a limited use case I think it's fine to assume it's an array.

**Pull request merged**